android-core/src/main/java/com/mparticle/identity/IdentityApiRequest.java

```diff
@@ -211,4 +214,47 @@ public Builder userAliasHandler(@Nullable UserAliasHandler userAliasHandler) {
             return this;
         }
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true; // Check if the same object
+        if (obj == null || getClass() != obj.getClass()) return false; // Check for null and class match
+
+        IdentityApiRequest that = (IdentityApiRequest) obj; // Cast to IdentityApiRequest
+
+        // Compare all relevant fields
+        return Objects.equals(userIdentities, that.userIdentities) &&
+                Objects.equals(otherOldIdentities, that.otherOldIdentities) &&
+                Objects.equals(otherNewIdentities, that.otherNewIdentities) &&
+                Objects.equals(userAliasHandler, that.userAliasHandler) &&
+                Objects.equals(mpid, that.mpid);
+    }
+
+    @NonNull
+    @Override
+    public String toString() {
+        return "userIdentities"+userIdentities+" otherOldIdentities " +otherOldIdentities+" otherNewIdentities "+otherNewIdentities
+                +" userAliasHandler "+userAliasHandler+" mpid "+String.valueOf(mpid);
+    }
+
+    public String convertString(){
+        return "";
+    }
+
+    public String objectToHash() {
+        String input =this.toString();
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
+            byte[] hashBytes = md.digest(input.getBytes());
+            StringBuilder hexString = new StringBuilder();
+            for (byte b : hashBytes) {
+                String hex = Integer.toHexString(0xff & b);
+                if (hex.length() == 1) hexString.append('0');
+                hexString.append(hex);
+            }
+            return hexString.toString().substring(0, 16); // Shorten to first 16 characters
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
Here I see we throw the exception instead of swallowing it. Can we also have a fallback here? Also this is duplicating the same code as the `generateHash` method above, can we move that to a utility class and reuse it instead?